src/main/webapp/consentGroup/NewConsentGroup.js

```diff
@@ -233,22 +272,41 @@ class NewConsentGroup extends Component {
     }
   }
 
+  isConsentFormUploaded() {
+    let isConsentFormUploaded = false;
+    if (this.state.files !== null && this.state.files.length > 0 && 
+      this.state.files.filter(file => file.fileKey === 'Consent Document').length > 0) {
+        isConsentFormUploaded = true;
+    }
+    this.setState(prev => {
+      prev.isConsentFormPresent = isConsentFormUploaded;
+      if (isConsentFormUploaded) {
+        prev.generalDataFormData.noConsentFormReason = '';
+      }
+      return prev;
+    });
+    return isConsentFormUploaded;
+  }
+
   validateGeneralData(field) {
     let investigatorLastName = false;
     let institutionProtocolNumber = false;
     let consentGroupName = false;
     let collaboratingInstitution = false;
     let describeConsentGroup = false;
-    let requireMta = false;
     let institutionalSourcesName = false;
     let institutionalSourcesCountry = false;
-
+    let noConsentFormReason = false;
     let isValid = true;
+    
+    if (!this.state.isConsentFormPresent && isEmpty(this.state.generalDataFormData.noConsentFormReason)) {
+      noConsentFormReason = true;
+      isValid = false;
```

Comment:
There is an interesting pattern here. Instead of assigning and potentially re-assigning `isValid` many, many times, instead, it could be calculated once at the end of this sequence of conditional checks. For example, 
```
const isValid = !(noConsentFormReason || consentGroupName || investigatorLastName || institutionProtocolNumber || institutionProtocolNumber || requireMta || collaboratingInstitution || describeConsentGroup || institutionalSourcesName || institutionalSourcesCountry || institutionalSourcesName);
```
This refactoring is a bit outside the scope of the code here, but it might be worth looking into.